tools/scripts/broker_test_publisher.sh

```diff
@@ -0,0 +1,146 @@
+            ;;
+         d)
+            case $OPTARG in
+                queue)
+                    destination="QueueName"
+                    connection_factory_name="QueueConnectionFactory"
+                    jndi_file_location="resources/jndi_queue.properties"
+                ;;
+                topic)
+                    destination="TopicName"
+                    connection_factory_name="TopicConnectionFactory"
+                    jndi_file_location="resources/jndi_topic.properties"
+                    ;;
+                ?)
+                echo $OPTARG
+                    echo "$OPTARG is an invalid destination.JMS destination should be a queue or a topic"
+                    exit
+                ;;
+            esac
+            break
+            ;;
+         h)
+            help_text="Welcome to ballerina message broker micro-benchmark tool\n\nUsage:\n\t./broker_test_publisher.sh [command].\n\nCommands\n\t-h  ask for help\n\t-p  set the location of the properties file\n\t-d  set jms destination type queue/topic\n"
+            printf "$help_text"
+            exit
+            ;;
+         ?)
+            printf "Invalid command.Run ./broker_test_publisher.sh -h for usage.\n"
+            exit
+            ;;
+     esac
+done
+
+# check for target folder
+if [ ! -e target/ ];
+then
+    mkdir -p target/publisher
+fi
+
+# hash-map to store user desired parameters
+declare -A user_inputs=(["jmeter_home"]="" ["thread_count"]="1" ["number_of_messages"]="1000000" ["throughput"]="5000" ["message_size"]="10KB")
+
+# Method to extract values from property file
+getProperty()
+{
+    local property_key=$1
+    local user_value=`cat ${properties_file_location} | grep ${property_key} | cut -d'=' -f2`
+    echo "$user_value"
+}
+
+for parameter in "${!user_inputs[@]}";
+do
+    value=$(getProperty $parameter)
+    if [ "$value" != '' ]
+        then
+            user_inputs["$parameter"]="$value"
+    fi
+done
+
+# validate inputs
+if [ ${user_inputs["thread_count"]} == '0' ]  || [ ${user_inputs["throughput"]} == '0' ]
+    then
+        # Command invoked cannot execute
+        echo Thread count and/or throughput cannot be zero
+        exit
+    fi
+
+# Summarizing inputs
+echo Jmeter home is set to - ${user_inputs["jmeter_home"]}
+echo Starting test process with ${user_inputs["number_of_messages"]} messages and a and throughput - ${user_inputs["throughput"]}
+
+# calculate loop count
+loop_count=$(echo "${user_inputs["number_of_messages"]}/${user_inputs["thread_count"]}+1" | bc)
+duration_of_the_test=$(echo "${user_inputs["number_of_messages"]}/${user_inputs["throughput"]}" | bc)
+
+# variable to store message math
+text_message_file_location="sample_messages/1kB.json"
+case ${user_inputs["message_size"]} in
+	10KB)
+		text_message_file_location="sample_messages/10kB.json"
+		;;
+	100KB)
+        text_message_file_location="sample_messages/100kB.json"
+		;;
+	1MB)
+		text_message_file_location="sample_messages/1MB.json"
+		;;
```

Comment:
if the message size is none of these, we need to echo an error and exit.

Reply:
Done